Button to request to maintain a record

knirirr opened the pull request:
This is a draft whilst I deal with two issues relating to this ticket:
1. I need to work out how to do a nice popup modal (suggestions welcome) for when a request is successfully received by the API.
2. This ticket must be completed first: https://github.com/FAIRsharing/FAIRsharing-API/issues/175

knirirr committed:
First draft of code for #494.

knirirr committed:
Autogenerated documentation files

terazus commented:
Netlify builds are likely to fail due to console.log statements. Didn't verify though.

knirirr commented:
I always check for them. Might have missed one though.

terazus commented:
In Record.vue:
![image](https://user-images.githubusercontent.com/6538906/93871165-10250b80-fc83-11ea-9474-cec50ba494af.png)

knirirr committed:
Buttons for #494 appear functional.

knirirr committed:
Merge branch 'mt_request_ownership_494' of github.com:FAIRsharing/fairsharing.github.io into mt_request_ownership_494

knirirr committed:
Autogenerated documentation files

terazus commented on src/components/Client/RESTClient.js:
```diff
@@ -269,6 +269,31 @@ class RESTClient {
         return response.data;
     }
 
+    async claimRecord(recordID, userToken) {
```
Please add the JSdoc comments for these two methods.
I'm also realizing i forgot to add them for edit and create publications.

terazus commented on src/views/Records/Record.vue:
```diff
@@ -16,16 +16,25 @@
         >
           EDIT
         </v-btn>
+        <v-btn
```
Do we agree that if a user can both edit and claim, then two buttons would  show up?
You may want to use a v-else-if="mycondition" instead.
Not a request, just a thing to consider.

knirirr replied:
This is one I'm not completely sure about. Technically, curators don't need to claim anything, so I was thinking about allowing them to assign maintainer roles to themselves directly in an admin interface to be developed later. But, I also thought they might request that this button show up for them as well.

terazus commented on src/views/Records/Record.vue:
```diff
@@ -190,6 +201,31 @@
+              if (claim.error) {
+                _module.error = "Sorry, your request to claim this record failed. Please contact us.";
+                _module.canClaim = false;
+              } else {
```
can you please put else's on new lines ? :D
The official name is Stroustrup styling.
Here in case you are interested: https://eslint.org/docs/rules/brace-style

My reasoning behind this is that webstorm has issues dealing with collapsing/expanding blocks styled with **one true brace**. It works much better with **Stroustrup**.

terazus commented on src/views/Records/Record.vue:
```diff
@@ -190,6 +201,31 @@
+                const claim = await client.canClaim(recordID, _module.user().credentials.token);
+                if (claim.error) {
+                  _module.canClaim = false;
+                } else {
```
same as above

terazus reviewed (changes requested):
A very few minors things to changes:
- [x] missing JSdoc comments 
- [x] Stroustrup styling

knirirr committed:
Changed if/else styling for
https://github.com/FAIRsharing/fairsharing.github.io/pull/496/files/b19e756213b763554429d0f97bc7b0f005dbbbb0#

knirirr committed:
Autogenerated documentation files

knirirr committed:
Merge branch 'dev' into mt_request_ownership_494

knirirr committed:
Autogenerated documentation files

knirirr committed:
Added some docstrings.

knirirr committed:
Merge branch 'mt_request_ownership_494' of github.com:FAIRsharing/fairsharing.github.io into mt_request_ownership_494

knirirr committed:
Autogenerated documentation files

knirirr committed:
Docstrings for RESTClient.

knirirr committed:
Merge branch 'mt_request_ownership_494' of github.com:FAIRsharing/fairsharing.github.io into mt_request_ownership_494

knirirr committed:
Autogenerated documentation files
